index.ts

```diff
@@ -48,85 +46,36 @@ export interface LogHandler {
   (data: LogData): void
 }
 
-/**
- * The global log event bus from which all events are emitted
- * and handlers are attached.
- *
- * When in Node, exposes the event API of Node `process`.
- * When in a browser, creates adaptor functions to mimic the
- * Node API using `window` event handling functions.
- */
-let bus: {
-  emit: (event: string, data: LogData) => void
-  listeners: (event: string) => LogHandler[]
-  addListener: (event: string, handler: LogHandler) => void
-  removeListener: (event: string, handler: LogHandler) => void
-  removeAllListeners: (event: string) => void
-}
-if (typeof process !== 'undefined') {
-  bus = {
-    /* eslint-disable @typescript-eslint/unbound-method */
-    emit: process.emit as typeof bus.emit,
-    listeners: process.listeners as typeof bus.listeners,
-    addListener: process.addListener as typeof bus.addListener,
-    removeListener: process.removeListener,
-    removeAllListeners: process.removeAllListeners,
-    /* eslint-enable @typescript-eslint/unbound-method */
-  }
-}
-/* istanbul ignore next */
-if (typeof window !== 'undefined') {
-  /**
-   * To mimic the Node event API in the browser it is necessary to:
-   *
-   * - wrap `LogData` in a `CustomEvent` when emitting an event and
-   *   unwrap it when handling an event
-   * - maintain a list of event listeners (`window` does not expose
-   *   that for us)
-   * - use a map of handlers to listeners so that we can remove them
-   */
-  type CustomEventListener = (customEvent: CustomEvent<LogData>) => void
-  const listeners = new Map<LogHandler, CustomEventListener>()
-  bus = {
-    emit: (event: string, data: LogData) => {
-      window.dispatchEvent(
-        new CustomEvent<LogData>(event, { detail: data })
-      )
-    },
-    listeners: () => {
-      return Array.from(listeners.keys())
-    },
-    addListener: (event: string, handler: LogHandler) => {
-      const listener = (customEvent: CustomEvent<LogData>): void =>
-        handler(customEvent.detail)
-      // @ts-ignore
-      window.addEventListener(event, listener)
-      listeners.set(handler, listener)
-    },
-    removeListener: (event: string, handler: LogHandler) => {
-      const listener = listeners.get(handler)
-      if (listener === undefined) return
-      // @ts-ignore
-      window.removeEventListener(event, listener)
-      listeners.delete(handler)
-    },
-    removeAllListeners: (event: string) => {
-      Array.from(listeners.values()).map((listener) => {
-        // @ts-ignore
-        window.removeEventListener(event, listener)
-      })
-      listeners.clear()
-    },
+// Global `logga` instance
+
+const root =
+  typeof window !== 'undefined'
+    ? window
+    : typeof global !== 'undefined'
+    ? global
+    : // Ignore in coverage because is expected to be unreachable
+      // istanbul ignore next
+      {}
+
+function Logga(): LogHandler[] {
+  const name = '_logga'
+  if (name in root) {
+    // istanbul ignore next
+    // @ts-ignore
```

Comment:
It would be better if we extended the `global/Window` type definition instead, for correctness as well as avoiding having to repeatedly use ignore statements like on line 70.

Reply:
Would you be able to do that please?

Reply:
done. I've made the global `_logga` an optional, to force checking its existence when someone tries to use it.
Feel free to suggest changes, or merge this PR at your leisure @nokome

Reply:
Thanks! Will merge now.